### PR TITLE
Fix mech codes to stop AI abuse

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Mechs now copy the user mob state and factions when you enter them, making AI no longer ignore mechs
Remove the faction and mob state when you exit the mech so AI ignore it ones again.
Fixed it so mechs now eject dead users.

## Why / Balance
Abuse in mech code allow it to be used to ignore AI and attack it without ever getting hit back.

## How to test
Spawn in,
Spawn a mech
Enter the mech
Spawn any hostile AI
Die.

## Media
N/A

## Breaking changes
N/A

**Changelog**
N/A

### CODE REVIEW

**Comment by ErhardSteinhauer** [comment]
c# is outside of my expertise, but the .yml part I can review. And that single yml line is baller.

**Review by ErhardSteinhauer — APPROVED** [review]

File: Content.Server/Mech/Systems/MechSystem.cs [hunk 1]
```diff
@@ -35,6 +38,7 @@ public sealed partial class MechSystem : SharedMechSystem
     [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
     [Dependency] private readonly SharedPopupSystem _popup = default!;
     [Dependency] private readonly UserInterfaceSystem _ui = default!;
+    [Dependency] private readonly AppearanceSystem _appearance = default!; // Frontier
```
Comment (whatston3) [inline Content.Server/Mech/Systems/MechSystem.cs:41]: Unused, is this here for a reason?
```suggestion
```

File: Content.Server/Mech/Systems/MechSystem.cs [hunk 2]
```diff
@@ -228,6 +232,17 @@ private void OnMechEntry(EntityUid uid, MechComponent component, MechEntryEvent
             return;
         }
 
+        // Frontier - Make AI Attack mechs based on user.
+        if (TryComp<MobStateComponent>(args.User, out var state))
```
Comment (whatston3) [inline Content.Server/Mech/Systems/MechSystem.cs:236]: Question about style - state is unused, should be replaced with `_`?
```suggestion
        if (TryComp<MobStateComponent>(args.User, out var _))
```

File: Content.Shared/Mech/EntitySystems/SharedMechSystem.cs [hunk 3]
```diff
@@ -388,6 +390,14 @@ public bool TryEject(EntityUid uid, MechComponent? component = null)
         RemoveUser(uid, pilot);
         _container.RemoveEntity(uid, pilot);
         UpdateAppearance(uid, component);
+
+        // Frontier - Make NPC AI attack Mechs
+        if (TryComp<MobStateComponent>(uid, out var state))
+            RemComp<MobStateComponent>(uid);
+        if (TryComp<NpcFactionMemberComponent>(uid, out var faction))
+            RemComp<NpcFactionMemberComponent>(uid);
```
Comment (whatston3) [inline Content.Shared/Mech/EntitySystems/SharedMechSystem.cs:398]: Again, unused variables.
```suggestion
        if (TryComp<MobStateComponent>(uid, out var _))
            RemComp<MobStateComponent>(uid);
        if (TryComp<NpcFactionMemberComponent>(uid, out var _))
            RemComp<NpcFactionMemberComponent>(uid);
```

File: Content.Shared/NPC/Components/NpcFactionMemberComponent.cs [hunk 4]
```diff
@@ -2,10 +2,11 @@
 using Content.Shared.NPC.Systems;
 using Robust.Shared.GameStates;
 using Robust.Shared.Prototypes;
+using Content.Shared.Mech.EntitySystems; // Frontier
 
 namespace Content.Shared.NPC.Components;
 
-[RegisterComponent, NetworkedComponent, Access(typeof(NpcFactionSystem))]
+[RegisterComponent, NetworkedComponent, Access(typeof(NpcFactionSystem), typeof(SharedMechSystem))] // Frontier - Added MechSystem
```
Comment (whatston3) [inline Content.Shared/NPC/Components/NpcFactionMemberComponent.cs:9]: Does this even need an access component?  I can't find any other partial implementations, and all of the fields here are public.  Am I missing something?

**Review by whatston3 — CHANGES_REQUESTED** [review]
Works well under testing.

A few small suggestions, mostly unused variables.

Might be misunderstanding how AccessAttribute works, but is it really needed?

File: Content.Shared/NPC/Components/NpcFactionMemberComponent.cs (hunk 4 above)
Comment (dvir001) [reply Content.Shared/NPC/Components/NpcFactionMemberComponent.cs:9]: "Read Write only" error
I did need it to set factions

Comment (whatston3) [reply Content.Shared/NPC/Components/NpcFactionMemberComponent.cs:9]: Ah, gotcha.  Thank you.

**Review by whatston3 — APPROVED** [review]

**Pull request merged**